From dd0070222986d57a2c1fd50641fc2453394ed1b3 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:28 -0400
Subject: [infiniband] mlx4: Use actual number of PCI functions (PF + VFs) for alias GUID logic

Message-id: <cf34049bb296d8206203d171860f64d951da3115.1410425016.git.avadai@redhat.com>
Patchwork-id: 93727
O-Subject: [Patch RHEL7.1 V1 092/197] mlx4: Use actual number of PCI functions (PF + VFs) for alias GUID logic
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Matan Barak <matanb@mellanox.com>

The code which is dealing with SRIOV alias GUIDs in the mlx4 IB driver has some
logic which operated according to the maximal possible active functions (PF + VFs).

After the single port VFs code integration this resulted in a flow of false-positive
warnings going to the kernel log after the PF driver started the alias GUID work.

Fix it by referring to the actual number of functions.

Signed-off-by: Matan Barak <matanb@mellanox.com>
Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit e471b40321a94f07d13b8a9e4b064885cf08835d)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/mlx4/alias_GUID.c b/drivers/infiniband/hw/mlx4/alias_GUID.c
index 2f215b9..0eb141c 100644
--- a/drivers/infiniband/hw/mlx4/alias_GUID.c
+++ b/drivers/infiniband/hw/mlx4/alias_GUID.c
@@ -154,7 +154,7 @@ void mlx4_ib_notify_slaves_on_guid_change(struct mlx4_ib_dev *dev,
    continue;
 
   slave_id = (block_num * NUM_ALIAS_GUID_IN_REC) + i ;
-  if (slave_id >= dev->dev->num_slaves)
+  if (slave_id >= dev->dev->num_vfs + 1)
    return;
   tmp_cur_ag = *(__be64 *)&p_data[i * GUID_REC_SIZE];
   form_cache_ag = get_cached_alias_guid(dev, port_num,
-- 
1.7.1